lms/static/scripts/frontend_apps/components/DataLoader.tsx

```diff
@@ -13,8 +13,12 @@ export type DataLoaderProps<Data> = {
    */
   children: ComponentChildren;
 
-  /** Function that fetches data if {@link loaded} is false. */
-  load: () => Promise<Data>;
+  /**
+   * Function that fetches data if {@link loaded} is false.
+   * It provides an abort signal that gets aborted when the DataLoader is
+   * unmounted.
```

Comment:
It also gets aborted if the `onLoad` callback changes, which makes sense I think, in case something changed and the route decided to load different data.